NIPTool/models/nipt_results.py

```diff
@@ -0,0 +1,101 @@
+        Column("Chr12_Ratio", [CanConvertValidation(float)]),
+        Column("Chr14_Ratio", [CanConvertValidation(float)]),
+        Column("Chr15_Ratio", [CanConvertValidation(float)]),
+        Column("Chr16_Ratio", [CanConvertValidation(float)]),
+        Column("Chr17_Ratio", [CanConvertValidation(float)]),
+        Column("Chr19_Ratio", [CanConvertValidation(float)]),
+        Column("Chr20_Ratio", [CanConvertValidation(float)]),
+        Column("Chr22_Ratio", [CanConvertValidation(float)]),
+        Column("Chr1", [CanConvertValidation(float)]),
+        Column("Chr2", [CanConvertValidation(float)]),
+        Column("Chr3", [CanConvertValidation(float)]),
+        Column("Chr4", [CanConvertValidation(float)]),
+        Column("Chr5", [CanConvertValidation(float)]),
+        Column("Chr6", [CanConvertValidation(float)]),
+        Column("Chr7", [CanConvertValidation(float)]),
+        Column("Chr8", [CanConvertValidation(float)]),
+        Column("Chr9", [CanConvertValidation(float)]),
+        Column("Chr10", [CanConvertValidation(float)]),
+        Column("Chr11", [CanConvertValidation(float)]),
+        Column("Chr12", [CanConvertValidation(float)]),
+        Column("Chr13", [CanConvertValidation(float)]),
+        Column("Chr14", [CanConvertValidation(float)]),
+        Column("Chr15", [CanConvertValidation(float)]),
+        Column("Chr16", [CanConvertValidation(float)]),
+        Column("Chr17", [CanConvertValidation(float)]),
+        Column("Chr18", [CanConvertValidation(float)]),
+        Column("Chr19", [CanConvertValidation(float)]),
+        Column("Chr20", [CanConvertValidation(float)]),
+        Column("Chr21", [CanConvertValidation(float)]),
+        Column("Chr22", [CanConvertValidation(float)]),
+        Column("ChrX", [CanConvertValidation(float)]),
+        Column("ChrY", [CanConvertValidation(float)]),
+        Column("FF_Formatted", [CanConvertValidation(float)]),
+        Column("FFY", [CanConvertValidation(float)]),
+        Column("FFX", [CanConvertValidation(float)]),
+        Column("DuplicationRate", [CanConvertValidation(float)]),
+        Column("Bin2BinVariance", [CanConvertValidation(float)]),
+        Column("UnfilteredCNVcalls", [CanConvertValidation(float)]),
+        Column("CNVSegment", [InListValidation(["Found"])]),
+        Column("Flowcell", []),
+        Column("SequencingDate", []),
+        Column("Median_13", [CanConvertValidation(float)]),
+        Column("Median_18", [CanConvertValidation(float)]),
+        Column("Median_21", [CanConvertValidation(float)]),
+        Column("Median_X", [CanConvertValidation(float)]),
+        Column("Median_Y", [CanConvertValidation(float)]),
+        Column("Stdev_13", [CanConvertValidation(float)]),
+        Column("Stdev_18", [CanConvertValidation(float)]),
+        Column("Stdev_21", [CanConvertValidation(float)]),
+        Column("Stdev_X", [CanConvertValidation(float)]),
+        Column("Stdev_Y", [CanConvertValidation(float)]),
+    ]
+)
```

Comment:
I've never used pandas validator, so this might be a bit of an ignorant question... Does validate-pandas have any built-in logic for checking if field can be empty?

Reply:
Totally valid question. allow_empty=False is supposed to do this, but I spent half morning trying to get it work without progress, so I gave up. Couldn't find any good working examples online either. https://tmiguelt.github.io/PandasSchema/#column